src/main/java/it/gov/pagopa/rtd/ms/rtdmsingestor/model/Transaction.java

```diff
@@ -22,58 +24,88 @@
 public class Transaction {
 
   @CsvBindByPosition(position = 0)
+  @Pattern(regexp = "[a-zA-Z0-9]{5}", message = "ABI length must be 5 alphanumeric char")
   String acquirerCode;
 
   @CsvBindByPosition(position = 1)
+  @Pattern(regexp = "[0-9]{2}", message = "Operation type length must match [0-9]{2}")
   String operationType;
 
   @CsvBindByPosition(position = 2)
+  @Pattern(regexp = "[0-9]{2}", message = "Circuit type length must match [0-9]{2}")
   String circuitType;
 
   @CsvBindByPosition(position = 3)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,64}",
+      message = "HPAN length must be 64 alphanumeric alphanumeric char at max")
   String hpan;
 
+  //ISO8601
+  @NotNull
   @CsvBindByPosition(position = 4)
   @CsvDate(value = "yyyy-MM-dd'T'HH:mm:ss.SSSX")
   Date trxDate;
 
   @CsvBindByPosition(position = 5)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,255}",
+      message = "ID trx acquirer length must be 255 alphanumeric char at max")
   String idTrxAcquirer;
 
   @CsvBindByPosition(position = 6)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,255}",
+      message = "ID trx issuer length must be 255 alphanumeric char at max")
   String idTrxIssuer;
 
   @CsvBindByPosition(position = 7)
+  @Pattern(regexp = "[a-zA-Z0-9]{0,255}",
+      message = "Correlation ID length must be 255 alphanumeric char at max")
   String correlationId;
 
+  @NotNull
   @CsvBindByPosition(position = 8)
   BigDecimal totalAmount;
 
   @CsvBindByPosition(position = 9)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,3}",
+      message = "Currency length must be 3 alphanumeric char at max")
   String amountCurrency;
 
   @CsvBindByPosition(position = 10)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,255}",
+      message = "Acquirer ID length must be 255 alphanumeric char at max")
   String acquirerId;
 
   @CsvBindByPosition(position = 11)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,255}",
+      message = "Merchant ID length must be 255 alphanumeric char at max")
   String merchantId;
 
   @CsvBindByPosition(position = 12)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,255}",
+      message = "Terminal ID length must be 255 alphanumeric char at max")
   String terminalId;
 
   @CsvBindByPosition(position = 13)
+  @Pattern(regexp = "[0-9]{6}|[0-9]{8}",
+      message = "Correlation ID length must match [0-9]{6}|[0-9]{8}")
   String bin;
 
   @CsvBindByPosition(position = 14)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,5}", message = "MCC length must be 5 alphanumeric char at max")
   String mcc;
 
   @CsvBindByPosition(position = 15)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,50}",
```

Comment:
Maybe this regex should be more restrictive

Reply:
Thank you, I'll fix it in another PR.